lib/rules/dot-notation.js

```diff
@@ -70,27 +101,28 @@ module.exports = {
                                 propertyValue: JSON.stringify(node.property.value)
                             },
                             fix(fixer) {
-                                const leftBracket = sourceCode.getTokenAfter(node.object, astUtils.isOpeningBracketToken);
-                                const rightBracket = sourceCode.getLastToken(node);
-
-                                if (sourceCode.getFirstTokenBetween(leftBracket, rightBracket, { includeComments: true, filter: astUtils.isCommentToken })) {
-
-                                    // Don't perform any fixes if there are comments inside the brackets.
-                                    return null;
-                                }
-
-                                const tokenAfterProperty = sourceCode.getTokenAfter(rightBracket);
-                                const needsSpaceAfterProperty = tokenAfterProperty &&
-                                    rightBracket.range[1] === tokenAfterProperty.range[0] &&
-                                    !astUtils.canTokensBeAdjacent(String(node.property.value), tokenAfterProperty);
-
-                                const textBeforeDot = astUtils.isDecimalInteger(node.object) ? " " : "";
-                                const textAfterProperty = needsSpaceAfterProperty ? " " : "";
-
-                                return fixer.replaceTextRange(
-                                    [leftBracket.range[0], rightBracket.range[1]],
-                                    `${textBeforeDot}.${node.property.value}${textAfterProperty}`
-                                );
+                                return fixDotNotationWithValue(fixer, node, node.property.value);
+                            }
+                        });
+                    }
+                }
+                if (
```

Comment:
It seems like there's some duplicated code here relating to checking keywords, checking valid identifiers, and reporting. I think that could be avoided by doing something like this:
```js
function checkComputedProperty(node, value) {
    if (validIdentifier.test(value) && /* ... */ ) {
        // ...
    }
}

return {
    MemberExpression(node) {
        if (node.computed && node.property.type === "Literal") {
            checkComputedProperty(node, node.property.value);
        } else if (node.computed && node.property.type === "TemplateLiteral" && node.property.expressions.length === 0) {
            checkComputedProperty(node, node.property.quasis[0].value.cooked);
        }

    }
};
```

This would also make `fixDotNotationWithValue` unnecessary, since there would only be one `context.report` call in the file. In other words, this would extract all of the value-checking logic into a function rather than only extracting the autofix logic.

Reply:
Yeah agree thats cleaner. Will change now.